### ISSUE TITLE
With lein-garden-0.2.7: java.lang.RuntimeException: No such var: fs/absolute, compiling:(leiningen/garden.clj:44:13)

### ISSUE THREAD

**Issue by devurandom** [noprompt/lein-garden#41]
With `lein-garden 0.2.7`, every invokation of `lein` causes the following backtrace:
```
Error: problem requiring leiningen.garden/activate hook
clojure.lang.Compiler$CompilerException: java.lang.RuntimeException: No such var: fs/absolute, compiling:(leiningen/garden.clj:44:13)
 at clojure.lang.Compiler.analyze (Compiler.java:6688)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$InvokeExpr.parse (Compiler.java:3766)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6870)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$InvokeExpr.parse (Compiler.java:3834)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6870)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.access$300 (Compiler.java:38)
    clojure.lang.Compiler$LetExpr$Parser.parse (Compiler.java:6269)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6868)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$BodyExpr$Parser.parse (Compiler.java:6001)
    clojure.lang.Compiler$FnMethod.parse (Compiler.java:5380)
    clojure.lang.Compiler$FnExpr.parse (Compiler.java:3972)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6866)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.access$300 (Compiler.java:38)
    clojure.lang.Compiler$DefExpr$Parser.parse (Compiler.java:589)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6868)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler.eval (Compiler.java:6931)
    clojure.lang.Compiler.load (Compiler.java:7379)
    clojure.lang.RT.loadResourceScript (RT.java:372)
    clojure.lang.RT.loadResourceScript (RT.java:363)
    clojure.lang.RT.load (RT.java:453)
    clojure.lang.RT.load (RT.java:419)
    clojure.core$load$fn__5677.invoke (core.clj:5893)
    clojure.core$load.invokeStatic (core.clj:5892)
    clojure.core$load.doInvoke (core.clj:5876)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.core$load_one.invokeStatic (core.clj:5697)
    clojure.core$load_one.invoke (core.clj:5692)
    clojure.core$load_lib$fn__5626.invoke (core.clj:5737)
    clojure.core$load_lib.invokeStatic (core.clj:5736)
    clojure.core$load_lib.doInvoke (core.clj:5717)
    clojure.lang.RestFn.applyTo (RestFn.java:142)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$load_libs.invokeStatic (core.clj:5774)
    clojure.core$load_libs.doInvoke (core.clj:5758)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$require.invokeStatic (core.clj:5796)
    clojure.core$require.doInvoke (core.clj:5796)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    leiningen.core.utils$require_resolve.invokeStatic (utils.clj:89)
    leiningen.core.utils$require_resolve.invoke (utils.clj:82)
    leiningen.core.project$load_hook$fn__6227.invoke (project.clj:719)
    leiningen.core.project$load_hook.invokeStatic (project.clj:719)
    leiningen.core.project$load_hook.invoke (project.clj:718)
    leiningen.core.project$load_hooks.invokeStatic (project.clj:737)
    leiningen.core.project$load_hooks.doInvoke (project.clj:730)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    leiningen.core.project$activate_middleware.invokeStatic (project.clj:774)
    leiningen.core.project$activate_middleware.invoke (project.clj:770)
    leiningen.core.project$init_project.invokeStatic (project.clj:888)
    leiningen.core.project$init_project.invoke (project.clj:878)
    leiningen.core.project$read.invokeStatic (project.clj:956)
    leiningen.core.project$read.invoke (project.clj:953)
    leiningen.core.project$read.invokeStatic (project.clj:957)
    leiningen.core.project$read.invoke (project.clj:953)
    leiningen.core.main$_main$fn__5895.invoke (main.clj:397)
    leiningen.core.main$_main.invokeStatic (main.clj:394)
    leiningen.core.main$_main.doInvoke (main.clj:391)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.lang.Var.invoke (Var.java:379)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invokeStatic (core.clj:646)
    clojure.main$main_opt.invokeStatic (main.clj:314)
    clojure.main$main_opt.invoke (main.clj:310)
    clojure.main$main.invokeStatic (main.clj:421)
    clojure.main$main.doInvoke (main.clj:384)
    clojure.lang.RestFn.invoke (RestFn.java:436)
    clojure.lang.Var.invoke (Var.java:388)
    clojure.lang.AFn.applyToHelper (AFn.java:160)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
Caused by: java.lang.RuntimeException: No such var: fs/absolute
 at clojure.lang.Util.runtimeException (Util.java:221)
    clojure.lang.Compiler.resolveIn (Compiler.java:7138)
    clojure.lang.Compiler.resolve (Compiler.java:7108)
    clojure.lang.Compiler.analyzeSymbol (Compiler.java:7069)
    clojure.lang.Compiler.analyze (Compiler.java:6648)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$InvokeExpr.parse (Compiler.java:3766)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6870)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$InvokeExpr.parse (Compiler.java:3834)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6870)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.access$300 (Compiler.java:38)
    clojure.lang.Compiler$LetExpr$Parser.parse (Compiler.java:6269)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6868)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler$BodyExpr$Parser.parse (Compiler.java:6001)
    clojure.lang.Compiler$FnMethod.parse (Compiler.java:5380)
    clojure.lang.Compiler$FnExpr.parse (Compiler.java:3972)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6866)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6856)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.access$300 (Compiler.java:38)
    clojure.lang.Compiler$DefExpr$Parser.parse (Compiler.java:589)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:6868)
    clojure.lang.Compiler.analyze (Compiler.java:6669)
    clojure.lang.Compiler.analyze (Compiler.java:6625)
    clojure.lang.Compiler.eval (Compiler.java:6931)
    clojure.lang.Compiler.load (Compiler.java:7379)
    clojure.lang.RT.loadResourceScript (RT.java:372)
    clojure.lang.RT.loadResourceScript (RT.java:363)
    clojure.lang.RT.load (RT.java:453)
    clojure.lang.RT.load (RT.java:419)
    clojure.core$load$fn__5677.invoke (core.clj:5893)
    clojure.core$load.invokeStatic (core.clj:5892)
    clojure.core$load.doInvoke (core.clj:5876)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.core$load_one.invokeStatic (core.clj:5697)
    clojure.core$load_one.invoke (core.clj:5692)
    clojure.core$load_lib$fn__5626.invoke (core.clj:5737)
    clojure.core$load_lib.invokeStatic (core.clj:5736)
    clojure.core$load_lib.doInvoke (core.clj:5717)
    clojure.lang.RestFn.applyTo (RestFn.java:142)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$load_libs.invokeStatic (core.clj:5774)
    clojure.core$load_libs.doInvoke (core.clj:5758)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$require.invokeStatic (core.clj:5796)
    clojure.core$require.doInvoke (core.clj:5796)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    leiningen.core.utils$require_resolve.invokeStatic (utils.clj:89)
    leiningen.core.utils$require_resolve.invoke (utils.clj:82)
    leiningen.core.project$load_hook$fn__6227.invoke (project.clj:719)
    leiningen.core.project$load_hook.invokeStatic (project.clj:719)
    leiningen.core.project$load_hook.invoke (project.clj:718)
    leiningen.core.project$load_hooks.invokeStatic (project.clj:737)
    leiningen.core.project$load_hooks.doInvoke (project.clj:730)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    leiningen.core.project$activate_middleware.invokeStatic (project.clj:774)
    leiningen.core.project$activate_middleware.invoke (project.clj:770)
    leiningen.core.project$init_project.invokeStatic (project.clj:888)
    leiningen.core.project$init_project.invoke (project.clj:878)
    leiningen.core.project$read.invokeStatic (project.clj:956)
    leiningen.core.project$read.invoke (project.clj:953)
    leiningen.core.project$read.invokeStatic (project.clj:957)
    leiningen.core.project$read.invoke (project.clj:953)
    leiningen.core.main$_main$fn__5895.invoke (main.clj:397)
    leiningen.core.main$_main.invokeStatic (main.clj:394)
    leiningen.core.main$_main.doInvoke (main.clj:391)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.lang.Var.invoke (Var.java:379)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invokeStatic (core.clj:646)
    clojure.main$main_opt.invokeStatic (main.clj:314)
    clojure.main$main_opt.invoke (main.clj:310)
    clojure.main$main.invokeStatic (main.clj:421)
    clojure.main$main.doInvoke (main.clj:384)
    clojure.lang.RestFn.invoke (RestFn.java:436)
    clojure.lang.Var.invoke (Var.java:388)
    clojure.lang.AFn.applyToHelper (AFn.java:160)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
```

`project.clj` contains:
```edn
(defproject my-project "0.1.0-SNAPSHOT"
  :dependencies [[org.clojure/clojure "1.8.0"]
                 [org.clojure/clojurescript "1.7.228"]
                 [prismatic/plumbing "0.5.2"]
                 [org.omcljs/om "0.9.0"]
                 [om-widgets "0.3.0"]
                 [garden "1.3.2"]
                 [cljsjs/jquery "2.1.4-0"]
                 [jayq "2.5.4"]]

  :plugins [[lein-cljsbuild "1.1.3"]
            [lein-garden "0.2.7"]]

  :hooks [leiningen.cljsbuild
          leiningen.garden]

  :source-paths ["src/clj" "src/cljs"]

  :garden {:builds [{:id "main"
                     :source-paths ["src/styles"]
                     :stylesheet my.style/main
                     :compiler {:output-to "target/css/main.css"}}]}

  :cljsbuild {:builds {:debug {:source-paths ["src/cljs"]
                               :compiler {:main my.main
                                          :output-to "target/js/debug.js"
                                          :optimizations :none
                                          :source-map true}}
                       :release {:source-paths ["src/cljs"]
                                 :compiler {:main my.main
                                            :output-to "target/js/main.js"
                                            :optimizations :advanced
                                            :pretty-print false}}}})
```

Downgrading to `lein-garden 0.2.6` fixes the issue.

**Comment by devurandom** [comment]
Removing the hook and changing the `:plugins` line to the following:
```edn
[lein-garden "0.2.7" :exclusions [org.clojure/clojure org.apache.commons/commons-compress]]
```
allows me to run `lein deps :plugin-tree`:
```edn
 [lein-ancient "0.6.8" :exclusions [[org.clojure/clojure]]]
   [ancient-clj "0.3.11" :exclusions [[com.amazonaws/aws-java-sdk-s3]]]
     [clj-aws-s3 "0.3.10" :exclusions [[joda-time] [org.clojure/clojure] [com.amazonaws/aws-java-sdk]]]
     [clj-http "2.0.0" :exclusions [[joda-time] [org.clojure/clojure] [com.cognitect/transit-clj] [crouton] [slingshot]]]
       [commons-io "2.4" :exclusions [[org.clojure/clojure]]]
       [org.apache.httpcomponents/httpclient "4.5" :exclusions [[org.clojure/clojure]]]
       [org.apache.httpcomponents/httpcore "4.4.1" :exclusions [[org.clojure/clojure]]]
       [org.apache.httpcomponents/httpmime "4.5" :exclusions [[org.clojure/clojure]]]
     [clj-time "0.10.0" :exclusions [[joda-time] [org.clojure/clojure]]]
     [com.fasterxml.jackson.core/jackson-core "2.6.1" :exclusions [[joda-time] [org.clojure/clojure]]]
     [commons-codec "1.10" :exclusions [[joda-time] [org.clojure/clojure]]]
     [commons-logging "1.2" :exclusions [[joda-time] [org.clojure/clojure]]]
     [joda-time "2.8.2" :exclusions [[joda-time] [org.clojure/clojure]]]
     [org.clojure/data.xml "0.0.8" :exclusions [[joda-time] [org.clojure/clojure]]]
   [com.amazonaws/aws-java-sdk-s3 "1.9.0" :exclusions [[joda-time]]]
     [com.amazonaws/aws-java-sdk-core "1.9.0"]
       [com.fasterxml.jackson.core/jackson-annotations "2.1.1"]
       [com.fasterxml.jackson.core/jackson-databind "2.1.1"]
   [jansi-clj "0.1.0"]
     [org.fusesource.jansi/jansi "1.11"]
   [org.clojure/tools.reader "0.10.0"]
   [potemkin "0.4.1"]
     [clj-tuple "0.2.2"]
     [riddley "0.1.10"]
   [rewrite-clj "0.4.12"]
   [version-clj "0.1.2"]
 [lein-cljsbuild "1.1.3"]
   [fs "1.1.2"]
     [org.apache.commons/commons-compress "1.3"]
     [org.clojure/clojure "1.3.0"]
 [lein-garden "0.2.7" :exclusions [[org.clojure/clojure] [org.apache.commons/commons-compress]]]
   [garden "1.3.1"]
     [com.yahoo.platform.yui/yuicompressor "2.4.8" :exclusions [[rhino/js]]]
   [me.raynes/fs "1.4.4"]
   [ns-tracker "0.3.0"]
     [org.clojure/java.classpath "0.2.2"]
     [org.clojure/tools.namespace "0.2.10"]
 [lein-midje "3.2"]
```

`~/.lein/profiles.clj` contains:
```edn
{:user {:plugins [[lein-ancient "0.6.8" :exclusions [org.clojure/clojure]]
                  [lein-midje "3.2"]]}}
```

**Comment by noprompt** [comment]
@devurandom That seems odd. #35 was supposed to have fixed this problem.

**Comment by devurandom** [comment]
You predicted that this issue here would arise: https://github.com/noprompt/lein-garden/pull/35#issuecomment-108213124

**Comment by noprompt** [comment]
@devurandom I did, I'm an idiot. Yeah, this is a problem. I think we just need to get rid of the dependency on `fs`. It's much more trouble than it's worth.

**Comment by WorldsEndless** [comment]
I'll just second this. Just fired up the latest lein-garden 0.2.7 and had wishful thoughts for "lein garden auto" before receiving this error.

**Comment by noprompt** [comment]
I think this may have been fixed by #44 but I can't be sure. I'm sorry I can't be of more help right now!
